addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4256,16 +4254,19 @@ var SnippetsMenu = Widget.extend({
         }
         delete data._toMutex;
         ev.stopPropagation();
-        this._buttonClick((after) => this._execWithLoadingEffect(() => {
-            const oldOnFailure = data.onFailure;
-            data.onFailure = () => {
-                if (oldOnFailure) {
-                    oldOnFailure();
-                }
-                after();
-            };
-            this.trigger_up('request_save', data);
-        }, true), this.$el[0].querySelector('button[data-action=save]'));
+        this._buttonClick(async (after) => {
+            await this.postSnippetDropPromise;
+            this._execWithLoadingEffect(() => {
```

Comment:
- This was returned before, I think it should still be ?
- Are you sure awaiting the "snippet drop promise" cannot be done inside the `_execWithLoadingEffect` ?

Reply:
> Are you sure awaiting the "snippet drop promise" cannot be done inside the `_execWithLoadingEffect` ?

Awaiting the `postSnippetDropPromise` inside `_execWithLoadingEffect` would lead to a potential deadlock scenario. The mutex within `_execWithLoadingEffect` would wait for `postSnippetDropPromise` to resolve. However, `postSnippetDropPromise` relies on the completion of `callPostSnippetDrop`, which also runs on same mutex.